django/core/cache/backends/db.py

```diff
@@ -157,7 +162,19 @@ def _base_set(self, mode, key, value, timeout=DEFAULT_TIMEOUT):
                                 current_expires = converter(current_expires, expression, connection)
 
                     exp = connection.ops.adapt_datetimefield_value(exp)
-                    if result and (mode == 'set' or (mode == 'add' and current_expires < now)):
+                    if mode == 'touch':
```

Comment:
Perhaps simplify this to reduce the indentation (or equivalent):
```python
if result and mode == 'touch':
    ...
elif result and (mode == 'set' or (mode == 'add' and current_expires < now)):
    ...
elif mode != 'touch':
    ...
else:
    return False
```

Reply:
Thanks!